### DISCUSSION TITLE
From Apple: "Action required: Transition from XML to App Store Connect API"

### DISCUSSION THREAD

**Discussion by klep** [fastlane/fastlane#20490] [General] [answered]
Hi there.

We received the message below in an email from Apple this morning. Of course, it doesn't indicate which XML API(s) we're using or what third party tools might be using them. Behind the scenes, **does Fastlane still use any XML-based APIs?** My recollection is that there are some Enterprise features that don't have API support, so perhaps those are the culprit?

```
We noticed that you recently used the XML feed to manage and deliver content to App Store Connect. Starting in November 2022, the XML feed will no longer support in-app purchases, subscriptions, metadata, or app pricing. You can manage this content via the App Store Connect REST API, which makes it easy to customize and automate your workflows.

The XML feed will continue to support existing Game Center management functionality.If you have any questions, contact us.

Best regards,
Apple Developer Relations
```

**Comment by joshdholtz** [comment, marked as answer]
@klep _fastlane_ does not use any XML APIs so this shouldn't be an issue 😇  At least none that I've seen in the past 6 years! I wonder if its something related to iTMS transport? 🤔 

I didn't get any email but I'll see if I can either find a list of what's being deprecated or contact someone at Apple to let me know!

**Reply by klep** [reply]
Thanks for the quick reply, @joshdholtz. We'll look elsewhere for the culprit, or let it reveal itself in November :)

**Comment by 404pilot** [comment]
I am using `fastlane deliver` to upload pkg file to apple connect. It looks like `deliver` is using itunes transporter (https://github.com/fastlane/fastlane/blob/2851a3d537231a59164e73c96f69c56af683cff8/fastlane_core/lib/fastlane_core/itunes_transporter.rb#L572)

Does transporter still use "XML feed" feature? what is the alternative?

**Comment by hborders** [comment]
@milch mentioned transporter could be affected in response to my tweet about this: https://twitter.com/acrooow/status/1550171241801080832

**Comment by hborders** [comment]
This is a related thread on the Apple Developer Forums: https://developer.apple.com/forums/thread/708235

**Comment by hborders** [comment]
This Apple News post says it's about `in-app purchases, subscriptions, metadata, and app pricing`:

> The App Store Connect REST API lets you customize and automate tasks across developer tools, giving you greater flexibility and efficiency in your workflows. Starting in November 2022, you’ll need to use this API instead of the XML feed to automate management of in-app purchases, subscriptions, metadata, and app pricing. The XML feed will continue to support existing Game Center management functionality.

https://developer.apple.com/news/?id=yqf4kgwb
